libs/algo/odc/algo/_percentile.py

```diff
@@ -0,0 +1,58 @@
+from typing import List, Sequence, Tuple
+import dask.array as da
+import xarray as xr
+import numpy as np
+from ._geomedian import reshape_for_geomedian
+from ._masking import keep_good_np
+
+
+def np_percentile(xx, percentile, nodata):
+
+    mask = (xx != nodata) & (~np.isnan(xx))
```

Comment:
Let's support 2 modes here.
1. Int + Nodata
2. Float with nan and no "nodata"

For int case convert to `float32` with one of the existing methods call itself with that then convert back to original `dtype/nodata`.

For floating point inputs `assert nodata is None` and assume `NaN` sort higher and just keep using current method.

Alternatively only support `float32` on input and perform conversion `int->float32->int` as a Dask operation and hope those things will get fused.

Reply:
@robbibt points out on Slack that xarray already has this operation in Dask form, but also, if that is trying to be too smart and do some non-trivial parallel sorting there is also `np.nanpercentile?` which is probably faster then all this, so we just need to write `int+nodata` wrapper for that.

Reply:
(copying from slack to keep a record)

unfortunately the numpy and  xr_array (which I think just wraps numpy ) are ~700x slower than this implementation!!